rustls-platform-verifier/src/tests/mod.rs

```diff
@@ -61,3 +61,10 @@ pub(crate) fn verification_time() -> pki_types::UnixTime {
     // Monday, March 11, 2024 8:30:25 PM UTC
     pki_types::UnixTime::since_unix_epoch(Duration::from_secs(1_710_189_025))
 }
+
+fn ensure_global_state() {
```

Comment:
I think the `cfg(test)` guard can be outside the function?

Reply:
++ on this. We shouldn't need the guard at all because the `tests` module should be under the same thing already.

Reply:
That's what I thought, too. However, the `tests` module is gated by `#[cfg(any(test, feature = "ffi-testing"))]`, and whatever builds it with `ffi-testing` in CI isn't a test build, so it doesn't get the *ring* dev-dependency, so the build fails. I think this is also related to the remaining failure, in which the android tests presumably don't set up a default provider. Haven't actually found the android tests or worked out a strategy to address that yet. Maybe we need some sort of automatic fallback like `rustls` seems to have.

Reply:
That looks like it it actually, good catch. With the improved Android logging I saw it was throwing the missing crypto provider error for every test. Locally I updated the `ffi-testing` feature and made sure to call `ensure_global_state` in the Android tests. After that everything worked. Here's the changes if you'd like to commit them:
```diff
diff --git a/rustls-platform-verifier/Cargo.toml b/rustls-platform-verifier/Cargo.toml
index cdffcdf..dc2f369 100644
--- a/rustls-platform-verifier/Cargo.toml
+++ b/rustls-platform-verifier/Cargo.toml
@@ -19,7 +19,7 @@ crate-type = ["cdylib", "rlib"]
 # Enables a C interface to use for testing where `cargo` can't be used.
 # This feature is not stable, nor is the interface exported when it is enabled.
 # Do not rely on this or use it in production.
-ffi-testing = ["android_logger"]
+ffi-testing = ["android_logger", "rustls/ring"]
 # Enables APIs that expose lower-level verifier types for debugging purposes.
 dbg = []
 # Enables `log::debug` base64-encoded logging of all end-entity certificates processed
diff --git a/rustls-platform-verifier/src/tests/ffi.rs b/rustls-platform-verifier/src/tests/ffi.rs
index bf18a01..6a4a5af 100644
--- a/rustls-platform-verifier/src/tests/ffi.rs
+++ b/rustls-platform-verifier/src/tests/ffi.rs
@@ -47,6 +47,7 @@ mod android {
                     .with_filter(log_filter),
             );
             crate::android::init_hosted(env, cx).unwrap();
+            crate::tests::ensure_global_state();
             std::panic::set_hook(Box::new(|info| {
                 let msg = if let Some(msg) = info.payload().downcast_ref::<&'static str>() {
                     msg
diff --git a/rustls-platform-verifier/src/tests/mod.rs b/rustls-platform-verifier/src/tests/mod.rs
index 63e6490..16ff02a 100644
--- a/rustls-platform-verifier/src/tests/mod.rs
+++ b/rustls-platform-verifier/src/tests/mod.rs
@@ -62,9 +62,6 @@ pub(crate) fn verification_time() -> pki_types::UnixTime {
     pki_types::UnixTime::since_unix_epoch(Duration::from_secs(1_710_189_025))
 }
 
-fn ensure_global_state() {
-    #[cfg(test)]
-    {
-        _ = rustls::crypto::ring::default_provider().install_default();
-    }
+pub(crate) fn ensure_global_state() {
+    let _ = rustls::crypto::ring::default_provider().install_default();
 }
```